### ISSUE TITLE
Question with weighted sample

### ISSUE THREAD

**Issue by yifanxie** [dmlc/xgboost#775]
Dear XGB team:

Once again thanks for developing and maintaining this excellent package, and it has immense contribute to my adventure on Kaggle competition. I am not a specialist in machine learning algorithm, but am really keen to be a better "operator" of XGB, so I hope it is ok to ask a few technical question here that is related to weighted sample. 

How does the "weight" keyword in DMatrix function in a multiple classification problem? It seems that most of the sample script I can find that use the "weight" keyword is to do with binary classification such as the Higgs Boson challenge. 

So for example, if I have a dataset with 7000 instances of class "0", 1500 instances of class "1", and 400 instances of class "2":
-  Can I use the "weight" keyword to help me "balance" the dataset? I am using mlogloss as the cost function. 
- In the higgs-cv.py file you perform rescaling of the weight on train and test set for each cv fold. Can this be done in R with the same "fprepoc" keyword?  

Thanks for your answer! 

Best Regards
Yifan

**Comment by tqchen** [comment]
The weight is presented per-instance, and you can also use the weight to specific weight of each instance for multi-class classification
